libs/langchain-azure-cosmosdb/src/azure_cosmosdb_mongodb_vcore.ts

```diff
@@ -0,0 +1,483 @@
+import {
+  ObjectId,
+  Collection,
+  Document as MongoDBDocument,
+  MongoClient,
+  Db,
+  Filter,
+} from "mongodb";
+import type { EmbeddingsInterface } from "@langchain/core/embeddings";
+import {
+  MaxMarginalRelevanceSearchOptions,
+  VectorStore,
+} from "@langchain/core/vectorstores";
+import { Document, DocumentInterface } from "@langchain/core/documents";
+import { maximalMarginalRelevance } from "@langchain/core/utils/math";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+
+/** Azure Cosmos DB for MongoDB vCore Similarity type. */
+export const AzureCosmosDBMongoDBvCoreSimilarityType = {
+  /** Cosine similarity */
+  COS: "COS",
+  /** Inner - product */
+  IP: "IP",
+  /** Euclidian distance */
+  L2: "L2",
+} as const;
+
+/** Azure Cosmos DB for MongoDB vCore Similarity type. */
+export type AzureCosmosDBMongoDBvCoreSimilarityType =
+  (typeof AzureCosmosDBMongoDBvCoreSimilarityType)[keyof typeof AzureCosmosDBMongoDBvCoreSimilarityType];
+
+/** Azure Cosmos DB for MongoDB vCore Index Options. */
+export type AzureCosmosDBMongoDBvCoreIndexOptions = {
+  /** Skips automatic index creation. */
+  readonly skipCreate?: boolean;
+  /** Number of clusters that the inverted file (IVF) index uses to group the vector data. */
+  readonly numLists?: number;
+  /** Number of dimensions for vector similarity. */
+  readonly dimensions?: number;
+  /** Similarity metric to use with the IVF index. */
+  readonly similarity?: AzureCosmosDBMongoDBvCoreSimilarityType;
+};
+
+/** Azure Cosmos DB for MongoDB vCore delete Parameters. */
+export type AzureCosmosDBMongoDBvCoreDeleteParams = {
+  /** List of IDs for the documents to be removed. */
+  readonly ids?: string | string[];
+  /** MongoDB filter object or list of IDs for the documents to be removed. */
+  readonly filter?: Filter<MongoDBDocument>;
+};
+
+/** Configuration options for the `AzureCosmosDBMongoDBvCoreVectorStore` constructor. */
+export interface AzureCosmosDBMongoDBvCoreConfig {
+  readonly client?: MongoClient;
+  readonly connectionString?: string;
+  readonly databaseName?: string;
+  readonly collectionName?: string;
+  readonly indexName?: string;
+  readonly textKey?: string;
+  readonly embeddingKey?: string;
+  readonly indexOptions?: AzureCosmosDBMongoDBvCoreIndexOptions;
+}
+
+/**
+ * Azure Cosmos DB for MongoDB vCore vector store.
+ * To use this, you should have both:
+ * - the `mongodb` NPM package installed
+ * - a connection string associated with a MongoDB VCore Cluster
+ *
+ * You do not need to create a database or collection, it will be created
+ * automatically.
+ *
+ * You also need an index on the collection, which is by default be created
+ * automatically using the `createIndex` method.
+ */
+export class AzureCosmosDBMongoDBvCoreVectorStore extends VectorStore {
```

Comment:
This is a very intimidating name - how about just `CosmosDBvCoreVectorStore`

Reply:
I agree with the long name being intimidating 😅 
Though I'm not sure it can be much better here, since vCore is a *tier* of the CosmosDB *MongoDB* API. So vCore without MongoDB might conflict with another API tier.

`AzureCosmosDBMongoDBVectorStore` might be the shortest we could do, as supporting a different tier could eventually be done with an option in case we need it. WDYT?

Reply:
@jacoblee93 I've renamed everything to `AzureCosmosDBMongoDBVectorStore`, not the best but still slightly better, and coherent with the other integration. Hope it's fine like this?

Reply:
Sure! Generally up to you, just wanted to raise it